corehq/apps/reports/views.py

```diff
@@ -1166,26 +1182,39 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
             "enddate": date_range.get("enddate") if date_range else "",
         })
 
-    return render(request, "reports/report_email.html", {
+    return render(request, "reports/report_email_content.html", {
         "reports": report_outputs,
         "domain": domain,
         "couch_user": owner_id,
         "DNS_name": get_url_base(),
         "owner_name": couch_user.full_name or couch_user.get_email(),
         "email": email,
         "notes": notes,
-        "unsub_link": get_url_base() + reverse('notification_unsubscribe', kwargs={
-            'scheduled_report_id': report_id, 'domain': domain}) if report_id else None,
         "report_type": _("once off report") if once else _("scheduled report"),
-    }), excel_attachments
+    }).content, excel_attachments
+
+
+def render_full_report_notification(request, content, email=None, report=None):
+    from dimagi.utils.web import get_url_base
+
+    unsub_link = None
+    if report and email:
+        unsub_link = get_url_base() + reverse('notification_unsubscribe', kwargs={
+            'scheduled_report_id': report._id,
+            'user_email': email,
+            'scheduled_report_secret': ReportNotification.get(report._id).get_secret(email)
```

Comment:
Is it necessary to fetch the notification from the DB again? Can't you just call `report.get_secret(email)`?